core/notifier/src/lib.rs

```diff
@@ -85,6 +77,17 @@ impl<C: Collection> NotifierInterface<C> for Notifier<C> {
     }
 }
 
+#[derive(Default, Clone)]
+pub struct EpochChangeNotificationsEmitter {
+    emitter: Arc<Notify>,
+}
+
+impl EpochNotifierEmitter for EpochChangeNotificationsEmitter {
+    fn epoch_changed(&self) {
+        self.emitter.notify_waiters()
+    }
+}
```

Comment:
following the comment above about getting rid of the injection flow with affair, we should use this event emitter instead to perform the notification.